### ISSUE TITLE
Missing reflection metadata of anonymous records 

### ISSUE THREAD

**Issue by Zaid-Ajaj** [fable-compiler/Fable#1815]
### Description

Reflection metadata of anonymous records are not properly resovled that same way as records are.  Would be a *really* nice to have for deserialization and protocol definitions with Fable.Remoting

### Repro code

```fs
open FSharp.Reflection

let anonRecordFields() = 
    let recordType = typeof<{| name: string |}>
    let fields = FSharpType.GetRecordFields recordType
    fields 

anonRecordFields()
|> Array.iter (fun field -> printfn "%s" field.Name)
```
compile to
```js
import { name, getRecordElements, obj } from "fable-library/Reflection.js";
import { toConsole, printf } from "fable-library/String.js";
export function anonRecordFields() {
  const recordType = obj;
  const fields = getRecordElements(recordType);
  return fields;
}
anonRecordFields().forEach(function action(field) {
  toConsole(printf("%s"))(name(field));
});
```
See `recordType` is compiled down to `obj` which causes an exception:
```
Uncaught Error: System.Object is not an F# record type
```
### Related information

* Fable version: repl
* Operating system: N/A

**Comment by alfonsogarciacaro** [comment]
Yes, I noticed that the other day to it. It shouldn't be difficult to implement, I'll do it :)

**Comment by alfonsogarciacaro** [comment]
At the end it was a bit trickier than expected but it should work now. Released in fable-compiler (npm) 2.3.3 :+1:

**Comment by Zaid-Ajaj** [comment]
@alfonsogarciacaro This is amazing, it is working really well and it is very nice for deserialization without having to define 100s of records, here is an example:
```fs
testCase "Deserializing JSON to anonymous records with lists" <| fun test ->
    """ { "numbers": [1,2,3,4,5]  } """
    |> Json.parseNativeAs<{| numbers: int list |}>
    |> fun result -> result.numbers 
    |> List.sum   
    |> test.areEqual 15
```
I've added a couple of [tests](https://github.com/Zaid-Ajaj/Fable.SimpleJson/commit/cefa168d2dd8434ed770558184552cddb4c0c63f) to SimpleJson and everything passes, I will let you know if I find something that isn't working, thanks a lot 🙏
